src/mail-app/contacts/model/NativeContactsSyncManager.ts

```diff
@@ -125,53 +133,84 @@ export class NativeContactsSyncManager {
 		return this.deviceConfig.getUserSyncContactsWithPhonePreference(this.loginController.getUserController().userId) ?? false
 	}
 
-	enableSync() {
+	async enableSync() {
+		const loginUsername = this.loginController.getUserController().loginUsername
+		const contactListId = await this.contactModel.getContactListId()
+		if (contactListId == null) return false
+		const contacts = await this.entityClient.loadAll(ContactTypeRef, contactListId)
+		const structuredContacts = contacts.map((c) => this.toStructuredContact(c))
+		await this.mobileContactsFacade.syncContacts(loginUsername, structuredContacts)
 		this.deviceConfig.setUserSyncContactsWithPhonePreference(this.loginController.getUserController().userId, true)
+		await this.askToDedupeContacts(structuredContacts)
+	}
+
+	/**
+	 * Check if syncing contacts is possible/allowed right now.
+	 *
+	 * On Android, this method simply requests permission to access contacts. On iOS, this also checks iCloud sync, as
+	 * it can interfere with
+	 */
+	async canSync(): Promise<boolean> {
+		if (!isApp()) {
+			throw new ProgrammingError("Can only check Contact permissions on app")
+		}
+
+		const isContactPermissionGranted = await locator.systemPermissionHandler.requestPermission(PermissionType.Contacts, "allowContactReadWrite_msg")
+		if (!isContactPermissionGranted) {
+			return false
+		}
+
+		return !isIOSApp() || this.checkIfExternalCloudSyncOnIos()
+	}
+
+	/**
+	 * Check that we are allowed to sync contacts on an iOS device
+	 * @returns false if no permission or iCloud sync is enabled and the user cancelled, or true if permission is granted and iCloud sync is disabled (or the user bypassed the warning dialog)
+	 */
+	private async checkIfExternalCloudSyncOnIos(): Promise<boolean> {
+		assert(isIOSApp(), "Can only check cloud syncing on iOS")
+
+		let localContactStorage = await this.mobileContactsFacade.isLocalStorageAvailable()
+		if (!localContactStorage) {
+			const choice = await Dialog.choice<"cancel" | "settings" | "enable">("externalContactSyncDetectedWarning_msg", [
```

Comment:
this is not translated and maybe also does not fit in dialog buttons, we should check it again